File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  result_arr = []
  string_arr = str.split("")
  len = str.size

  string_arr.each_with_index do |char, index|
    new_index = len - index
    result_arr[new_index] = char
  end

  result_arr.join
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string("hi")

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string("catbaby")

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  const resultArr = []
  const stringArr = str.split('')
  const length = str.length
  for (let i = 0; i < stringArr.length; i++) {
    const char = stringArr[i];
    newIndex = length - i 
    resultArr[newIndex] = char
  }
  return resultArr.join('')
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("Expecting: 'racecar'");
  console.log("=>", reverseString("racecar"));

  console.log("Expecting: 'rallodpoop'");
  console.log("=>", reverseString("poopdollar"));
}

module.exports = reverseString;

// Please add your pseudocode to this file

// iterate through string make into array
// length - index equals the letters new index
// And a written explanation of your solution
